### PR TITLE
Deduplicate pivot notifications in CLI runs

### PR DESCRIPTION
**Pull request by TJZine**
## Summary
- track per-run pivot messages when rendering screenshots so repeated notifications only log once
- document the streamlined pivot logging behaviour in the geometry pipeline notes
- add a regression test that exercises the CLI flow and asserts duplicate pivot notes are coalesced

## Testing
- pytest tests/test_frame_compare.py::test_run_cli_coalesces_duplicate_pivot_logs

------
https://chatgpt.com/codex/tasks/task_e_68f69fe8900883219ddbcc95a0569d72

## Summary by CodeRabbit

* **Bug Fixes**
  * Duplicate pivot notifications are now suppressed within a single CLI run so each pivot scenario is logged only once for clearer console output.

* **Documentation**
  * Geometry pipeline docs updated to describe the pivot message deduplication behavior.

* **Tests**
  * Added test coverage validating pivot message coalescing during CLI execution.

* **Chores**
  * Public export surface expanded to expose an additional analysis type.

### CODE REVIEW

File: tests/test_frame_compare.py [hunk 1]
```diff
@@ -1772,6 +1772,136 @@ def fake_upload(
...
+
+        def __init__(self, *args: object, **kwargs: object) -> None:
+            super().__init__(*args, **kwargs)
+
+        def log(self, *objects: object, **_kwargs: object) -> None:  # type: ignore[override]
+            message = objects[0] if objects else ""
+            RecordingConsole.pivot_logs.append(str(message))
+
+    RecordingConsole.pivot_logs = []
+    monkeypatch.setattr(frame_compare, "Console", RecordingConsole)
```
[the remark this answers is not in the document]
Comment (TJZine) [reply tests/test_frame_compare.py:1873]: @codex review and fix comments if relevant. suggested prompt: In tests/test_frame_compare.py around lines 1862 to 1873, the
RecordingConsole.pivot_logs is declared as a mutable class attribute without
ClassVar, which confuses type checkers; change its annotation to use
typing.ClassVar[list[str]] (e.g. pivot_logs: ClassVar[list[str]] = []) and
ensure ClassVar is imported from typing at the top of the file if not already
present.

**Comment by TJZine** [comment]
@codex investigate github CI failure. Failure output: "_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
    class RecordingConsole(base_console):  # type: ignore[misc]
>       pivot_logs: ClassVar[list[str]] = []
                    ^^^^^^^^
E       NameError: name 'ClassVar' is not defined
tests/test_frame_compare.py:1863: NameError
=========================== short test summary info ============================
FAILED tests/test_frame_compare.py::test_run_cli_coalesces_duplicate_pivot_logs - NameError: name 'ClassVar' is not defined
1 failed, 187 passed in 6.02s"
